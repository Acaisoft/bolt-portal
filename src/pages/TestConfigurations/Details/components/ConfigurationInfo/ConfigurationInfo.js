import React from 'react'
import { Typography, withStyles } from '@material-ui/core'

import styles from './ConfigurationInfo.styles'

const testSourceProperties = {
  repository: ['name', 'url'],
  test_creator: ['name'],
}

export const ConfigurationInfo = ({ configuration, classes }) => {
  const {
    test_source = {},
    configuration_type = {},
    name,
    configuration_parameters = [],
  } = configuration
  const { source_type } = test_source

  return (
    <div className={classes.root}>
      <Typography variant="body1">Name: {name}</Typography>
      <Typography variant="body1">Test type: {configuration_type.name}</Typography>
      <br />
      <Typography variant="body1">Test source: {source_type}</Typography>
      {source_type &&
        (testSourceProperties[source_type] || []).map(property => (
          <Typography key={property} variant="body2">
            {property}: {test_source[source_type][property]}
          </Typography>
        ))}
      <br />
      <Typography variant="body1">Configuration parameters:</Typography>
      {configuration_parameters.map(param => (
        <Typography key={param.parameter_slug} variant="body2">
          {param.parameter_slug}: {param.value}
        </Typography>
      ))}
    </div>
  )
}

export default withStyles(styles)(ConfigurationInfo)
