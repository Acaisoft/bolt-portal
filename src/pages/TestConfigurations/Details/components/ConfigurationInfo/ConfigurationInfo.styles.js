export default ({ palette, spacing, typography }) => {
  return {
    root: {},
  }
}
