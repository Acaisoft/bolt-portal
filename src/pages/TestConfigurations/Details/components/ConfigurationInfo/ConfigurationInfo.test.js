import React from 'react'
import { shallow } from 'enzyme'
import { mocks } from '~utils/tests'

import { ConfigurationInfo } from './ConfigurationInfo'

const initConfigurationInfo = overrides => {
  const mockProps = {
    classes: mocks.ClassesProxy,
    configuration: {
      name: 'Mocked configuration name',
      configuration_parameters: [
        { parameter_slug: 'mockedParam1', value: 'mockedValued1' },
        { parameter_slug: 'mockedParam2', value: 'mockedValued2' },
      ],
      configuration_type: {
        name: 'mockedConfigurationTypeName',
      },
      test_source: {
        source_type: 'repository',
        repository: {
          name: 'mockedRepositoryName',
          url: 'mockedRepostiroyUrl',
        },
      },
    },
  }
  const wrapper = shallow(<ConfigurationInfo {...mockProps} {...overrides} />)
  return { mockProps, wrapper }
}

describe('global: ConfigurationInfo', () => {
  it('renders without crashing', () => {
    const { wrapper } = initConfigurationInfo()
    expect(wrapper).toBeTruthy()
  })

  it('should render as expected', () => {
    const { wrapper } = initConfigurationInfo({})
    expect(wrapper).toMatchSnapshot()
  })
})
