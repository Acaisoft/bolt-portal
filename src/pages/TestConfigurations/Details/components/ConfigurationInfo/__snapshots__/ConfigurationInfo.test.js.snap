// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`global: ConfigurationInfo should render as expected 1`] = `
<div
  className="root"
>
  <TypographyMock
    variant="body1"
  >
    Name: 
    Mocked configuration name
  </TypographyMock>
  <TypographyMock
    variant="body1"
  >
    Test type: 
    mockedConfigurationTypeName
  </TypographyMock>
  <br />
  <TypographyMock
    variant="body1"
  >
    Test source: 
    repository
  </TypographyMock>
  <TypographyMock
    key="name"
    variant="body2"
  >
    name
    : 
    mockedRepositoryName
  </TypographyMock>
  <TypographyMock
    key="url"
    variant="body2"
  >
    url
    : 
    mockedRepostiroyUrl
  </TypographyMock>
  <br />
  <TypographyMock
    variant="body1"
  >
    Configuration parameters:
  </TypographyMock>
  <TypographyMock
    key="mockedParam1"
    variant="body2"
  >
    mockedParam1
    : 
    mockedValued1
  </TypographyMock>
  <TypographyMock
    key="mockedParam2"
    variant="body2"
  >
    mockedParam2
    : 
    mockedValued2
  </TypographyMock>
</div>
`;
